packages/data/src/world/mq/ganon_castle_mq.yml

```diff
@@ -35,92 +39,254 @@
+    GANON_CASTLE_MQ_FOREST_WIND_OPEN_EYE: "can_hit_triggers_distance"
   locations:
-    "MQ Ganon Castle Light Trial Chest": "soul_lizalfos_dinalfos && soul_enemy(SOUL_ENEMY_TORCH_SLUG) && can_play_zelda && has_weapon"
-    "MQ Ganon Pot Light End 1": "soul_lizalfos_dinalfos && soul_enemy(SOUL_ENEMY_TORCH_SLUG) && has_lens && can_hookshot && small_keys_ganon(3)"
-    "MQ Ganon Pot Light End 2": "soul_lizalfos_dinalfos && soul_enemy(SOUL_ENEMY_TORCH_SLUG) && has_lens && can_hookshot && small_keys_ganon(3)"
-    "MQ Ganon Castle Heart Light 1": "small_keys_ganon(2) && soul_lizalfos_dinalfos && soul_enemy(SOUL_ENEMY_TORCH_SLUG) && can_play_zelda && has_weapon"
-    "MQ Ganon Castle Heart Light 2": "small_keys_ganon(2) && soul_lizalfos_dinalfos && soul_enemy(SOUL_ENEMY_TORCH_SLUG) && can_play_zelda && has_weapon"
-"Ganon Castle Forest":
+    "MQ Ganon Castle Forest Trial First Chest": "event(GANON_CASTLE_MQ_FOREST_WIND_OPEN_EYE)"
+    "MQ Ganon Castle Forest Trial Second Chest": "event(GANON_CASTLE_MQ_FOREST_WIND_FROZEN_EYE)"
+"Ganon Castle Forest Trial Wind Room Ending Side":
   dungeon: Ganon
+  exits:
+    "Ganon Castle Forest Trial Wind Room Starting Side": "hookshot_anywhere || (has_hover_boots && climb_anywhere) || (can_longshot && event(GANON_CASTLE_MQ_FOREST_WIND_OPEN_EYE))"
+    "Ganon Castle Forest Trial Ending Room": "event(GANON_MQ_FOREST_SWITCH)"
   events:
-    GANON_TRIAL_FOREST: "soul_enemy(SOUL_ENEMY_STALFOS) && (can_play_time || can_play_elegy) && can_use_light_arrows"
+    GANON_CASTLE_MQ_FOREST_WIND_FROZEN_EYE: "has_fire"
+    GANON_CASTLE_MQ_FOREST_WIND_OPEN_EYE: "can_hit_triggers_distance"
+    GANON_MQ_FOREST_SWITCH: "can_play_time || can_play_elegy"
   locations:
-    "MQ Ganon Castle Forest Trial Key": "can_hookshot"
-    "MQ Ganon Castle Forest Trial First Chest": "soul_enemy(SOUL_ENEMY_STALFOS) && has_weapon && can_hit_triggers_distance"
-    "MQ Ganon Castle Forest Trial Second Chest": "soul_enemy(SOUL_ENEMY_STALFOS) && has_weapon && has_fire"
-    "MQ Ganon Pot Forest End 1": "soul_enemy(SOUL_ENEMY_STALFOS) && (can_play_time || can_play_elegy)"
-    "MQ Ganon Pot Forest End 2": "soul_enemy(SOUL_ENEMY_STALFOS) && (can_play_time || can_play_elegy)"
-"Ganon Castle Fire":
+    "MQ Ganon Castle Forest Trial Second Chest": "event(GANON_CASTLE_MQ_FOREST_WIND_FROZEN_EYE)"
+"Ganon Castle Forest Trial Ending Room":
   dungeon: Ganon
+  exits:
+    "Ganon Castle Forest Trial Wind Room Ending Side": "true"
   events:
-    GANON_TRIAL_FIRE: "can_use_light_arrows && has_tunic_goron_strict && (can_longshot || has_hover_boots) && silver_rupees_ganon_fire"
+    GANON_TRIAL_FOREST: "can_use_light_arrows"
+  locations:
+    "MQ Ganon Pot Forest End 1": "true"
+    "MQ Ganon Pot Forest End 2": "true"
+"Ganon Castle Fire Trial Entrance Ledge":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Main Room": "true"
+    "Ganon Castle Fire Trial Ending Ledge": "has_tunic_goron_strict && (can_longshot || has_hover_boots)"
```

Comment:
You can hookshot anywhere after lifting the monolith on the left side. Moreover, Hover Boots and Longshot Anywhere do not actually need strict either. Normal Longshot will also need the Silver Rupees collected as those are what makes the target appear.

Reply:
Added and changed.